lib/ymlr/encoder.ex

```diff
@@ -105,7 +105,18 @@ defmodule Ymlr.Encoder do
   end
 
   defp encode_as_io_list(data, _) when is_atom(data), do: Atom.to_string(data)
-  defp encode_as_io_list(data, level) when is_binary(data) do
+  defp encode_as_io_list(data, level) when is_binary(data), do: encode_binary(data, level)
+
+  defp encode_as_io_list(data, _) when is_number(data), do: "#{data}"
+
+  defp encode_as_io_list(data, _), do: raise(ArgumentError, message: "The given data #{inspect(data)} cannot be converted to YAML.")
+
+  defp encode_map_key(data) when is_atom(data), do: [Atom.to_string(data), ":"]
+  defp encode_map_key(data) when is_binary(data), do: [encode_binary(data, nil), ":"]
+  defp encode_map_key(data) when is_number(data), do: "#{data}:"
```

Comment:
Now that the number is in its binary representation, should it not go thrpugh encode_binary as well?

Reply:
No... guess not...